Foundation/NSLock.swift

```diff
@@ -258,3 +302,59 @@ open class NSCondition: NSObject, NSLocking {
     
     open var name: String?
 }
+
+private func timeSpecFrom(date: Date) -> timespec? {
+    let interval = date.timeIntervalSince1970
+    guard interval >= 0 else {
+        return nil
+    }
+
+    var ts = timespec()
+    ts.tv_sec = Int(interval.rounded(.down))
+    ts.tv_nsec = Int((interval - Double(ts.tv_sec)) * 1_000_000_000.0)
```

Comment:
this looks rather dangerous. If for floating point reasons `ts.tv_nsec` will be `1_000_000_000` or larger, all the timed wait functions will just return `EINVAL` IIRC.

Why not something similar to

```
private func timeSpecFrom(date: Date) -> timespec {
        guard interval >= 0 else { fatalError ("current time negative") }
        let nsecPerSec: Int64 = 1000000000
        let timeoutNS = Int64(interval * Double(nsecPerSec))

        var timeoutAbs = timespec(tv_sec: Int((timeoutNS / nsecPerSec)),
                                  tv_nsec: Int(timeoutNS % nsecPerSec))
        assert(timeoutAbs.tv_nsec >= 0 && timeoutAbs.tv_nsec < Int(nsecPerSec))
        assert(timeoutAbs.tv_sec >= curTime.tv_sec)
        return timeoutAbs
}
```